parsers/KPX.py

```diff
@@ -196,115 +188,132 @@ def get_long_term_prod_data(
         "_csrf": cookies_dict["XSRF-TOKEN"],
     }
 
-    res = r.post(LONG_TERM_PRODUCTION_URL, payload)
+    logger.debug(f"Fetching production data from {LONG_TERM_PRODUCTION_URL}")
+    res = session.post(LONG_TERM_PRODUCTION_URL, payload)
 
     assert res.status_code == 200
 
-    all_data = []
-
-    soup = BeautifulSoup(res.text, "html.parser")
-    table_rows = soup.find_all("tr")[1:]
-
-    for row in table_rows:
-
-        sanitized_date = [value[:-1] for value in row.find_all("td")[0].text.split(" ")]
-        curr_prod_datetime_string = (
-            "-".join(sanitized_date[:3]) + "T" + ":".join(sanitized_date[3:]) + ":00"
+    production_list = ProductionBreakdownList(logger)
+    events = _parse_long_term_prod_data(res.text)
+    for dt, event in events.items():
```

Comment:
As explained above here you could directly return the ProductionBreakdownList.

Reply:
Same